platform-sdk/docs/proposals/template.md

```diff
@@ -0,0 +1,109 @@
+# Summary
+
+Give a 1-3 sentence summary of the capability in the design proposal.
+
+| Metadata           | Entities                            | 
+|--------------------|-------------------------------------|
+| Last Updated       | 2024-06-07  (Edward)                |
+| Designers          | John, Mary, alpha-team              |
+| Functional Impacts | Services, DevOps, Mirror Node, etc. |
+| Related Proposals  | Proposal-1, Proposal-2              |
+| HIPS               | HIP-1, HIP-2,                       |
+
+# Purpose and Context
+
+Why is the design proposal necessary? Describe the background and purpose of the design proposal at a high level.
+
+## Dependencies, Interactions, and Implications
+
+What are the dependencies and pre-requisites of this proposal?  
+What other proposals, capabilities, and teams are impacted by this proposal?
+What does the proposal enable once it is completed?
+
+Remove this section if not applicable.
+
+# Requirements
+
+Describe the requirements and acceptance criteria that the design proposal must satisfy.
+
+# Design Decisions
+
+Describe the high level options available, the trade-offs between them and the decisions made.
+
+Remove this section if not applicable.
+
+# Changes
+
+## Public API
+
+Describe any public API changes or additions. Include stakeholders of the API.
```

Comment:
I would give examples of what constitutes a public API:

- Anything defined in protobuf
- Any functional API available for use outside of the module that provides it
- Anything written to disk

Reply:
I added the examples you provided.